lib/src/facts/linux/networking_resolver.cc

```diff
@@ -142,6 +142,17 @@ namespace facter { namespace facts { namespace linux {
         auto parse_route_line = [&known_route_types](string& line, std::vector<route>& routes) {
             vector<boost::iterator_range<string::iterator>> parts;
             boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+
+            // remove trailing "onlink" or "pervasive" flags
+            for (;;) {
+                std::string last_token(parts.back().begin(),
+                                       parts.back().end());
+                if (last_token == "onlink" || last_token == "pervasive")
+                    parts = decltype(parts)(parts.begin(), parts.end() - 1);
```

Comment:
Can we just use `parts.erase(parts.end()-1)` here to avoid copying the entire list?